crowbar_framework/app/models/proposal.rb

```diff
@@ -87,6 +92,26 @@ def raw_data=(value)
 
   private
 
+  # XXX: we need to be careful to not create an endless loop
+  def update_corresponding_proposal_object
+    proposal_object = ProposalObject.find_proposal_by_id(self.key)
+    if proposal_object
+      if proposal_object.raw_data != self.raw_data
+        proposal_object.raw_data = self.raw_data
+        proposal_object.save(sync: false)
```

Comment:
Hrm, don't we end up with some discrepancy in the revision between Proposal and ProposalObject (since it'll be bumped in this save again)?